app_dart/lib/src/service/cache_service.dart

```diff
@@ -49,40 +53,70 @@ class CacheService {
   Future<Uint8List?> getOrCreate(
     String subcacheName,
     String key, {
-    int attempt = 1,
-    Future<Uint8List> Function()? createFn,
+    required Future<Uint8List> Function()? createFn,
     Duration ttl = const Duration(minutes: 1),
   }) async {
     final Cache<Uint8List> subcache = cache.withPrefix(subcacheName);
     Uint8List? value;
 
+    const RetryOptions r = RetryOptions(
+      maxAttempts: maxCacheGetAttempts,
+      delayFactor: Duration(milliseconds: 50),
+    );
+
     try {
-      value = await subcache[key].get();
+      await r.retry(
+        () async {
+          value = await subcache[key].get();
+        },
+      );
     } catch (e) {
-      if (attempt < maxCacheGetAttempts) {
-        return getOrCreate(
-          subcacheName,
-          key,
-          attempt: ++attempt,
-          createFn: createFn,
-          ttl: ttl,
-        );
-      } else {
-        // Give up on trying to get the value from the cache.
-        value = null;
-      }
+      // If the last retry is unsuccessful on an exception we do not want to die
+      // here.
+      value = null;
     }
 
     // If given createFn, update the cache value if the value returned was null.
     if (createFn != null && value == null) {
       // Try creating the value
       value = await createFn();
-      await set(subcacheName, key, value, ttl: ttl);
+      await set(
+        subcacheName,
+        key,
+        value,
+        ttl: ttl,
+      );
     }
 
     return value;
   }
 
+  /// This method is the same as the [getOrCreate] method above except that it
+  /// enforces locking access.
+  ///
+  /// Note: these methods are intended to prevent issues around race conditions
```

Comment:
It would be good call out this only locks on this instance, not the entire fleet.

Reply:
Done.